scripts/opencontrail-install/contrail_install.rb

```diff
@@ -0,0 +1,324 @@
+end
+
+# Install contrail compute software
+def install_contrail_software_compute
+    contrail_rpms = [
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/python-contrail-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/python-contrail-vrouter-api-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-vrouter-utils-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-vrouter-init-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-lib-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-vrouter-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-vrouter-agent-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/noarch/contrail-setup-3.0-4100.fc21.noarch.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/noarch/contrail-vrouter-common-3.0-4100.fc21.noarch.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-vrouter-init-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-utils-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/x86_64/contrail-nodemgr-3.0-4100.fc21.x86_64.rpm",
+        "#{@ws}/contrail/controller/build/package-build/RPMS/noarch/contrail-vrouter-common-3.0-4100.fc21.noarch.rpm",
+    ]
+    sh("yum -y install #{contrail_rpms.join(" ")}", true)
+end
+
+# Return interface IP address, mask and gateway information
+def get_intf_ip(intf = @intf)
+    prefix = sh("ip addr show dev #{@intf}|\grep -w inet | " +
+                "\grep -v dynamic | awk '{print $2}'")
+    error("Cannot retrieve #{@intf}'s IP address") if prefix !~ /(.*)\/(\d+)$/
+    ip = $1
+    mask = IPAddr.new(prefix).inspect.split("/")[1].chomp.chomp(">")
+    gw = sh(%{netstat -rn |\grep "^0.0.0.0" | awk '{print $2}'})
+
+    return ip, mask, gw
+end
+
+# Provision contrail-vrouter agent and vrouter kernel module
+def provision_contrail_compute
+    ip, mask, gw = get_intf_ip
+    ifcfg = <<EOF
+#Contrail vhost0
+DEVICE=vhost0
+ONBOOT=yes
+BOOTPROTO=none
+IPV6INIT=no
+USERCTL=yes
+IPADDR=#{ip}
+NETMASK=#{mask}
+NM_CONTROLLED=no
+#NETWORK MANAGER BUG WORKAROUND
+SUBCHANNELS=1,2,3
+GATEWAY=#{gw}
+DNS1=8.8.8.8
+#DOMAIN="contrail.juniper.net. juniper.net. jnpr.net. contrail.juniper.net"
+EOF
+    File.open("/etc/sysconfig/network-scripts/ifcfg-vhost0", "w") { |fp|
+        fp.puts(ifcfg)
+    }
+
+    sh("sed 's/__DEVICE__/#{@intf}/' /etc/contrail/agent_param.tmpl > /etc/contrail/agent_param")
+    sh("sed -i 's/# type=kvm/type=kvm/' /etc/contrail/contrail-vrouter-agent.conf")
+    sh("/sbin/ifconfig eth1 |\grep ether | awk '{print $2}' | xargs echo > /etc/contrail/default_pmac")
```

Comment:
After digging, I find that problem is due to left-over code in
provisioning. I have raised
https://bugs.launchpad.net/juniperopenstack/+bug/1444660 to track the
cleanup necessary. For now, I just do "touch /etc/contrail/default_pmac" to
keep vnagent_ExecStartPre.sh not to exit.

Regards,
Ananth

On Wed, Apr 15, 2015 at 11:12 AM, Pedro Marques <notifications@github.com>
wrote:

> In scripts/opencontrail-install/contrail_install.rb
> <https://github.com/Juniper/contrail-kubernetes/pull/3#discussion_r28446519>
> :
>
> > +IPADDR=#{ip}
> > +NETMASK=#{mask}
> > +NM_CONTROLLED=no
> > +#NETWORK MANAGER BUG WORKAROUND
> > +SUBCHANNELS=1,2,3
> > +GATEWAY=#{gw}
> > +DNS1=8.8.8.8
> > +#DOMAIN="contrail.juniper.net. juniper.net. jnpr.net. contrail.juniper.net"
> > +EOF
> > +    File.open("/etc/sysconfig/network-scripts/ifcfg-vhost0", "w") { |fp|
> > +        fp.puts(ifcfg)
> > +    }
> > +
> > +    sh("sed 's/__DEVICE__/#{@intf}/' /etc/contrail/agent_param.tmpl > /etc/contrail/agent_param")
> > +    sh("sed -i 's/# type=kvm/type=kvm/' /etc/contrail/contrail-vrouter-agent.conf")
> > +    sh("/sbin/ifconfig eth1 |\grep ether | awk '{print $2}' | xargs echo > /etc/contrail/default_pmac")
>
> use "cat /sys/class/net/eth1/address" instead of ifconfig.
> We shouldn't be using this default_pmac intermediate file at all since it
> is simpler to cat the device address.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Juniper/contrail-kubernetes/pull/3/files#r28446519>.
>